pilot/pkg/model/xds_cache.go

```diff
@@ -183,25 +212,30 @@ func (l *lruCache) onEvict(k string, v cacheValue) {
 		xdsCacheEvictionsOnSize.Increment()
 	}
 
-	// The following cleanup logic needs to be called on every evict(whether passive or on exceeding size)
-	// because, passive eviction might be triggered by one of many dependent configs and we need to clear the
-	// reference from other dependents.
-	// We don't need to acquire locks, since this function is called when we write to the store.
-	l.clearIndexes(k, v)
+	// async clearing indexes
+	l.evictQueue = append(l.evictQueue, evictKeyConfigs{k, v.dependentConfigs})
 }
 
 func (l *lruCache) updateConfigIndex(k string, dependentConfigs []ConfigHash) {
 	for _, cfg := range dependentConfigs {
 		sets.InsertOrNew(l.configIndex, cfg, k)
 	}
-	l.recordDependentConfigSize()
 }
 
 func (l *lruCache) clearConfigIndex(k string, dependentConfigs []ConfigHash) {
+	c, exists := l.store.Get(k)
+	if exists {
```

Comment:
can we add some test of this case? Or do we already have them? 

I am a bit worried due to the fixed 5s timer we may be missing coverage of this in the tests

Reply:
There is already tests, need to do some update to make It work for async clearing